### ISSUE TITLE
ruby/ruby@62ec5eb

### ISSUE THREAD

**Issue by iliabylich** [whitequark/parser#545]
https://github.com/ruby/ruby/commit/62ec5eb01874472e2df9d871ef329d5a76b9c853

@whitequark Could you please confirm that it should be backported?

**Comment by whitequark** [comment]
This is a general quality improvement that should ideally be done, but our error messages on invalid inputs are not part of the API contract nor they, in general, match MRI.

**Comment by whitequark** [comment]
> I think it's even better to return a raw token name in the error message

I definitely disagree. Parser is a general Ruby frontend used as plumbing in many different tools. It should not spew internal token names in error messages and any instance where it does that is an oversight.

**Comment by iliabylich** [comment]
So, what's your verdict? Should it be implemented in parser?

**Comment by whitequark** [comment]
Yes. I was only saying that it isn't a *blocker*, and we do not need to match MRI's behavior or release train when introducing this.
